case-summarization/deploy/case-summarization.yaml

```diff
@@ -0,0 +1,450 @@
+                Resource:
+                  - Fn::Sub:
+                      - "arn:${AWS::Partition}:s3:::${DestinationBucket}"
+                      - DestinationBucket: {"Fn::ImportValue": "cid-DataCollection-Bucket"}
+        - PolicyName: "Bedrock-Access"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: "Allow"
+                Action:
+                  - "bedrock:ListFoundationModels"
+                  - "bedrock:InvokeModel"
+                Resource: "*" # cannot restrict this
+        - PolicyName: "SQS-Access"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: "Allow"
+                Action:
+                  - "sqs:SendMessage"
+                Resource: !GetAtt SummarizationDLQ.Arn
+              - Effect: "Allow"
+                Action:
+                  - "sqs:ReceiveMessage"
+                  - "sqs:DeleteMessage"
+                  - "sqs:GetQueueAttributes"
+                Resource: !GetAtt SummarizationQueue.Arn
+    Metadata:
+      cfn_nag:
+        rules_to_suppress:
+          - id: W11 # IAM role should not allow * resource on its permissions policy
+            reason: "cannot restrict ListFoundationModels and InvokeModel "
+
+  SummarizationLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      FunctionName: !Sub ${ResourcePrefix}support-case-summarization-Lambda
+      Description: "Lambda function to summarize aws support cases"
+      Runtime: python3.12
+      Architectures: [x86_64]
+      Code:
+        ZipFile: |
+          import os
+          import json
+          import logging
+
+          import boto3
+          from pydantic import BaseModel, Field
+          from llama_index.core.program import LLMTextCompletionProgram
+          from llama_index.llms.bedrock import Bedrock
+          from botocore.config import Config
+
+          logger = logging.getLogger(__name__)
+          logger.setLevel(getattr(logging, os.environ.get('LOG_LEVEL', 'INFO').upper(), logging.INFO))
+
+          REGION = os.environ.get("REGION", "us-east-1")
+          PROVIDER = os.environ.get("PROVIDER", "anthropic")
+          FOUNDATION_MODEL = os.environ.get("FOUNDATION_MODEL", "Claude 3.5 Sonnet")
+          INFERENCE_TYPE = os.environ.get("INFERENCE_TYPE", "ON_DEMAND")
+          TEMPERATURE = os.environ.get("TEMPERATURE", 1)
+          MAX_TOKENS = os.environ.get("MAX_TOKENS", 8096)
+          MAX_RETRIES = os.environ.get("MAX_RETRIES", 30)
+          TIMEOUT = os.environ.get("TIMEOUT", 60)
+          INSTRUCTIONS = os.environ.get("INSTRUCTIONS", '')
+          PROMPT_TEMPLATE = f"""
+          System: You are an expert technical writer specializing in creating concise, neutral summaries of AWS customers support interactions. Your task is to summarize conversations between customers and AWS Support, maintaining objectivity and clarity. Here is the Conversation to be summarized:
```

Comment:
Sensitive data is not normally seen in cases, but customers may still enter it in comments, right?

Any considerations around handling sensitive data here? Perhaps also making use of a support cases specific Guardrail with sensitive information filters?

Reply:
yes, there are guardrails just next to this CFN. 

Also a good point that because dashboard will show sensitive data without any filtering. So need to stress this in the documentation.